module/userProjects/helper.js

```diff
@@ -1048,323 +1213,352 @@ module.exports = class UserProjectsHelper {
+                        // create certificate object and add data if certificate template is present.
+                        if ( certificateTemplateDetails.length > 0 ) {
+                            projectCreation.data["certificate"] = _.pick(certificateTemplateDetails[0], ['templateUrl', 'status', 'criteria']);
+                            projectCreation.data["certificate"]["templateId"] = solutionDetails.certificateTemplateId;
+                        }
+                    }
                     
-            //         let getUserProfileFromObservation = false;
+                    let getUserProfileFromObservation = false;
     
-            //         if( bodyData && Object.keys(bodyData).length > 0 ) {
+                    if( bodyData && Object.keys(bodyData).length > 0 ) {
     
-            //             if( bodyData.hasAcceptedTAndC ) {
-            //                 projectCreation.data.hasAcceptedTAndC = bodyData.hasAcceptedTAndC;
-            //             }
+                        if( bodyData.hasAcceptedTAndC ) {
+                            projectCreation.data.hasAcceptedTAndC = bodyData.hasAcceptedTAndC;
+                        }
 
-            //             if( bodyData.referenceFrom ) {
-            //                 projectCreation.data.referenceFrom = bodyData.referenceFrom;
+                        if( bodyData.referenceFrom ) {
+                            projectCreation.data.referenceFrom = bodyData.referenceFrom;
                             
-            //                 if( bodyData.submissions ) {
-            //                     if ( bodyData.submissions.observationId && bodyData.submissions.observationId != "" ) {
-            //                         getUserProfileFromObservation = true;
-            //                     }
-            //                     projectCreation.data.submissions = bodyData.submissions;
-            //                 }
-            //             }
+                            if( bodyData.submissions ) {
+                                if ( bodyData.submissions.observationId && bodyData.submissions.observationId != "" ) {
+                                    getUserProfileFromObservation = true;
+                                }
+                                projectCreation.data.submissions = bodyData.submissions;
+                            }
+                        }
     
-            //             if( bodyData.role ) {
-            //                 projectCreation.data["userRole"] = bodyData.role;
-            //             }
+                        if( bodyData.role ) {
+                            projectCreation.data["userRole"] = bodyData.role;
+                        }
+
+                        if( 
+                            solutionDetails.entityType && bodyData[solutionDetails.entityType] 
+                        ) {
+                            let entityInformation = 
+                            await this.listByLocationIds(
```

Comment:
@Prajwal17Tunerlabs you can generate the query and pass it to entity service. no need of additional function to generate query and get data